proto/proto.go

```diff
@@ -28,22 +27,26 @@ func matchDomain(domain, pattern string) bool {
 	if pattern == "*" {
 		return true
 	}
-	if strings.HasPrefix(pattern, "*.") {
-		return strings.HasSuffix(domain, pattern[1:])
+
+	i := strings.Index(pattern, "*")
+	if i != -1 {
+		prefix := pattern[0:i]
+		suffix := pattern[i+1:]
```

Comment:
while this looks good,

https://pkg.go.dev/strings#Cut might be a little more readable

Reply:
Done. Thanks for the suggestion :+1: 

It looks much better